File: lib/py/uapi/internal/binary/EncodeBody.py
from typing import TYPE_CHECKING, cast

if TYPE_CHECKING:
    from uapi.internal.binary.BinaryEncoding import BinaryEncoding


def encode_body(message_body: dict[str, object], binary_encoding: 'BinaryEncoding') -> dict[object, object]:
    from uapi.internal.binary.EncodeKeys import encode_keys

    return cast(dict[object, object], encode_keys(message_body, binary_encoding))

File: lib/ts/uapi/internal/binary/EncodeBody.ts
import { BinaryEncoding } from 'uapi/internal/binary/BinaryEncoding';
import { encodeKeys } from 'uapi/internal/binary/EncodeKeys';

export function encodeBody(messageBody: Record<string, any>, binaryEncoding: BinaryEncoding): Record<any, any> {
    return encodeKeys(messageBody, binaryEncoding) as Record<any, any>;
}
